src/Components/Header/Header.js

```diff
@@ -1,11 +1,23 @@
-import React from 'react'
+import {useEffect} from 'react'
 import './Header.css'
 import logo from '../../assets/logo.png'
 import {Link , useNavigate}from 'react-router-dom'
-import {useAuth} from '../../Context' 
+import {useAuth , useVideos} from '../../Context' 
 function Header() {
    const navigate = useNavigate()
    const {token } = useAuth()
+   const {filterState , dispatchFilterState} = useVideos()
+   const { searchQuery } = filterState;
+   const searchHandler = (e) => {
+    if(location.pathname !== '/videos')
+        navigate("/videos");
+    else
+         dispatchFilterState({ type: "FILTER_BY_SEARCH", payload: e.target.value });
+  };
+  
+  useEffect(() => {
+    dispatchFilterState({ type: "FILTER_BY_SEARCH", payload: "" });
```

Comment:
Suggestion: you can pass the payload as object for better readability and prophylaxis.

Reply:
sure.Will do that